planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -2822,4 +2598,148 @@ void GoalPlannerModule::GoalPlannerData::update(
+    if (
+      modified_goal_opt && !goal_searcher->isSafeGoalWithMarginScaleFactor(
+                             modified_goal_opt.value(), hysteresis_factor, occupancy_grid_map,
+                             planner_data, static_target_objects)) {
+      next_state.state = PathDecisionState::DecisionKind::NOT_DECIDED;
+      return next_state;
+    }
+
+    // check current parking path collision
+    const auto & parking_path = pull_over_path.getParkingPath();
+    const std::vector<double> parking_path_curvatures = pull_over_path.getParkingPathCurvatures();
+    const double margin =
+      parameters.object_recognition_collision_check_hard_margins.back() * hysteresis_factor;
+    if (goal_planner_utils::checkObjectsCollision(
+          parking_path, parking_path_curvatures, static_target_objects, dynamic_target_objects,
+          planner_data->parameters, margin,
+          /*extract_static_objects=*/false, parameters.maximum_deceleration,
+          parameters.object_recognition_collision_check_max_extra_stopping_margin,
+          ego_polygons_expanded, true)) {
+      next_state.state = PathDecisionState::DecisionKind::NOT_DECIDED;
+      return next_state;
+    }
+
+    if (enable_safety_check && !next_state.is_stable_safe) {
+      next_state.state = PathDecisionState::DecisionKind::NOT_DECIDED;
+      return next_state;
+    }
+
+    // if enough time has passed since deciding status starts, transition to DECIDED
+    constexpr double check_collision_duration = 1.0;
+    const double elapsed_time_from_deciding = (now - prev_state_.stamp).seconds();
+    if (elapsed_time_from_deciding > check_collision_duration) {
+      next_state.state = PathDecisionState::DecisionKind::DECIDED;
+      return next_state;
+    }
+
+    // if enough time has NOT passed since deciding status starts, keep DECIDING
+    return next_state;
+  }
+
+  // if ego is sufficiently close to the start of the nearest candidate path, the path is decided
+  const auto & current_pose = planner_data->self_odometry->pose.pose;
+  const size_t ego_segment_idx =
+    autoware::motion_utils::findNearestSegmentIndex(current_path.points, current_pose.position);
+
+  const size_t start_pose_segment_idx = autoware::motion_utils::findNearestSegmentIndex(
+    current_path.points, pull_over_path.start_pose.position);
+  const double dist_to_parking_start_pose = calcSignedArcLength(
+    current_path.points, current_pose.position, ego_segment_idx, pull_over_path.start_pose.position,
+    start_pose_segment_idx);
+  if (dist_to_parking_start_pose > parameters.decide_path_distance) {
+    next_state.state = PathDecisionState::DecisionKind::NOT_DECIDED;
+    return next_state;
+  }
+
+  // if object recognition for path collision check is enabled, transition to DECIDING to check
+  // collision for a certain period of time. Otherwise, transition to DECIDED directly.
+  if (parameters.use_object_recognition) {
+    next_state.state = PathDecisionState::DecisionKind::DECIDED;
```

Comment:
@soblin 

![image](https://github.com/user-attachments/assets/67cd00dc-0577-4b67-a3ca-b2a3a6d780e0)

DECIDED is right?